tools/shared.py

```diff
@@ -199,6 +199,20 @@ def check_call(cmd, *args, **kw):
     exit_with_error("'%s' failed: %s", ' '.join(cmd), str(e))
 
 
+def get_all_files_under(dirname):
+  ret = []
+  for path, subdirs, files in os.walk(dirname):
+    for name in files:
+      ret.append(os.path.join(path, name))
```

Comment:
How about just 'yield os.path.join(path, name)` and then all this function just `all_files_under`

Reply:
Nice idea to use yield!